library/src/main/java/com/digi/xbee/api/WLANDevice.java

```diff
@@ -320,4 +352,457 @@ public void sendDataAsync(RemoteXBeeDevice remoteXBeeDevice, byte[] data)
 		// Not supported in WLAN modules.
 		throw new UnsupportedOperationException(OPERATION_EXCEPTION);
 	}
+	
+	/*
+	 * (non-Javadoc)
+	 * @see com.digi.xbee.api.AbstractXBeeDevice#addNetworkDataListener(com.digi.xbee.api.listeners.INetworkDataReceiveListener)
+	 */
+	@Override
+	public void addNetworkDataListener(INetworkDataReceiveListener listener) {
+		super.addNetworkDataListener(listener);
+	}
+	
+	/*
+	 * (non-Javadoc)
+	 * @see com.digi.xbee.api.AbstractXBeeDevice#removeNetworkDataListener(com.digi.xbee.api.listeners.INetworkDataReceiveListener)
+	 */
+	@Override
+	public void removeNetworkDataListener(INetworkDataReceiveListener listener) {
+		super.removeNetworkDataListener(listener);
+	}
+	
+	/**
+	 * Returns the protocol used in network transmissions.
+	 * 
+	 * @return The protocol used in network transmissions.
+	 * 
+	 * @see #setNetworkProtocol(NetworkProtocol)
+	 * @see com.digi.xbee.api.models.NetworkProtocol
+	 */
+	public NetworkProtocol getNetworkProtocol() {
+		return protocol;
+	}
+	
+	/**
+	 * Sets the network protocol used in network transmissions.
+	 * 
+	 * @param protocol The new network protocol to be set for network 
+	 *                 transmissions.
+	 * 
+	 * @see #getNetworkProtocol()
+	 * @see com.digi.xbee.api.models.NetworkProtocol
+	 */
+	public void setNetworkProtocol(NetworkProtocol protocol) {
+		if (protocol == null)
+			throw new NullPointerException("Network protocol cannot be null.");
+		
+		this.protocol = protocol;
+	}
+	
+	/**
+	 * Starts listening for incoming network transmissions in the provided port.
+	 * 
+	 * @param sourcePort Port to listen for incoming transmissions.
+	 * 
+	 * @throws TimeoutException if there is a timeout setting the source port.
```

Comment:
"if source port is out of range" as we don't know what those numbers might be in the future.